src/service/discovery.js

```diff
@@ -180,14 +183,17 @@ const fetchInitialDiscoveryTopics = async (limit = 50, page = 0) => {
  * @param {Number} page
  * @returns {FetchInitialDiscoveryUsersResponse}
  */
-const fetchInitialDiscoveryUsers = async (limit = 50, page = 0) => {
+const fetchInitialDiscoveryUsers = async (limit = 50, page = 0, isAnon = false) => {
   const body = {
     page
   };
 
   if (limit) body.limit = limit;
   try {
-    const response = await api.post(`discovery/init/user?limit=${limit}&page=${page}`, body);
+    const response = await api.post(
+      `discovery/init/user?limit=${limit}&page=${page}${isAnon ? '&allow_anon_dm=true' : ''}`,
+      body
+    );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, the `limit` and `page` parameters are directly interpolated into the URL string without any encoding or validation. This can potentially lead to security vulnerabilities such as injection attacks. Always validate and encode your parameters before using them in URLs.
```diff
- `discovery/init/user?limit=${limit}&page=${page}${isAnon ? '&allow_anon_dm=true' : ''}`,
+ `discovery/init/user?limit=${encodeURIComponent(limit)}&page=${encodeURIComponent(page)}${isAnon ? '&allow_anon_dm=true' : ''}`,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->